prog/weaveutil/datapath.go

```diff
@@ -67,3 +85,56 @@ func addDatapathInterface(args []string) error {
 	}
 	return odp.AddDatapathInterface(args[0], args[1])
 }
+
+// Helpers
+
+func checkMTU(dpname string, mtu int) bool {
+	var (
+		vpid   libodp.VportID
+		vpname string
+	)
+
+	// Create a dummy vxlan vport
+	for i := 0; i < 5; i++ {
+		portno, err := getUDPPortNo()
+		if err != nil {
+			return true
+		}
+		if vpid, vpname, err = odp.CreateVxlanVport(dpname, "vxlantest", portno); err == nil {
+			defer func() {
+				if err := odp.DeleteVport(dpname, vpid); err != nil {
+					fmt.Fprintf(os.Stderr, "unable to remove unused %q vxlan vport: %s\n", vpname, err)
+				}
+			}()
+			break
+		} else if errno, ok := err.(syscall.Errno); !(ok && errno == syscall.EADDRINUSE) {
+			return true
+		}
+	}
+	// Couldn't create the vport, skip the check
+	if vpname == "" {
+		return true
+	}
+
+	// Setting >1500 MTU on affected host' vport should fail with EINVAL
```

Comment:
My point is you need the condition.  With no condition it reads like you expect the failure every run.